tests/src/JIT/opt/FastTailCall/FastTailCallCandidates.cs

```diff
@@ -0,0 +1,1150 @@
+    {
+        int count = 0;
+        for (double i = 0; i < one; ++i)
+        {
+            if (i % 2 == 0)
+            {
+                ++count;
+            }
+        }
+
+        if (count == 1)
+        {
+            return 100;
+        }
+
+        else
+        {
+            return 108;
+        }
+    }
+
+    /// <summary>
+    /// Possible to fastTailCall only on arm64
+    /// </summary>
+    /// <remarks>
+    ///
+    /// This test case is really only interesting on arm64
+    ///
+    /// Arm64:
+    /// caller has 8 register arguments and 32 bytes of stack space
+    /// callee has 8 register arguments and 16 bytes of stack space
+    ///
+    /// x64 *nix:
+    /// caller has 8 register arguments and 32 bytes of stack space
+    /// callee has 8 register arguments and 16 bytes of stack space
+    ///
+    /// Arm64 can fastTailCall while x64 linux cannot
+    ///
+    /// Return 100 is a pass.
+    /// Return 108 is a failure.
+    ///
+    /// </remarks>
+    public static int CallerHFACaseWithStack(double one,
+                                             double two,
+                                             double three,
+                                             double four,
+                                             double five,
+                                             double six,
+                                             double seven,
+                                             double eight,
+                                             HFASize32 s1)
+    {
+        if (one % 2 == 0)
+        {
+            double a = one * 100;
+            double b = one + 1100;
+            return CalleeHFAStackSpace(one, 
+                                       two,
+                                       three,
+                                       four,
+                                       a,
+                                       b,
+                                       five,
+                                       six,
+                                       seven,
+                                       eight);
+        }
+        else
+        {
+            double b = one + 1599;
+            double a = one + 16;
+            return CalleeHFAStackSpace(one, 
+                                       two,
+                                       three,
+                                       four,
+                                       a,
+                                       b,
+                                       six,
+                                       five,
+                                       seven,
+                                       eight);
+        }
+    }
+
+    /// <summary>
+    /// Possible to fastTailCall only on arm64
+    /// </summary>
+    /// <remarks>
+    ///
+    /// This test case is really only interesting on arm64
+    ///
+    /// Arm64:
+    /// caller has 8 register arguments
+    /// callee has 8 register arguments
+    ///
+    /// x64 *nux:
```

Comment:
or just use "Linux" instead of "*nix" everywhere.

---
In reply to: [133306949](https://github.com/dotnet/coreclr/pull/12489#discussion_r133306949) [](ancestors = 133306949)